modules/custom/ymca_sync/src/SyncerRunner.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace Drupal\ymca_sync;
+
+use Drupal\Core\Config\ConfigFactory;
+use Drupal\Core\Lock\LockBackendInterface;
+use Drupal\Core\Logger\LoggerChannelInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Class SyncerRunner.
+ *
+ * @package Drupal\ymca_sync.
+ */
+class SyncerRunner {
+
+  /**
+   * The used lock backend instance.
+   *
+   * @var \Drupal\Core\Lock\LockBackendInterface
+   */
+  protected $lock;
+
+  /**
+   * LoggerChannelInterface.
+   *
+   * @var Drupal\Core\Logger\LoggerChannelInterface
+   *   LoggerChannelInterface.
+   */
+  protected $logger;
+
+  /**
+   * ConfigFactory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactory
+   */
+  protected $configFactory;
+
+  /**
+   * YMCA sync config.
+   *
+   * @var \Drupal\Core\Config\Config|\Drupal\Core\Config\ImmutableConfig
+   */
+  protected $config;
+
+
+  /**
+   * ContainerInterface.
+   *
+   * @var \Symfony\Component\DependencyInjection\ContainerInterface
+   *   ContainerInterface.
+   */
+  protected $container;
+
+  /**
+   * Implement construct method.
+   *
+   * @param \Drupal\Core\Lock\LockBackendInterface $lock
+   *   Lock.
+   * @param \Drupal\Core\Logger\LoggerChannelInterface $logger
+   *   LoggerChannelInterface.
+   * @param \Drupal\Core\Config\ConfigFactory $configFactory
+   *   ConfigFactory.
+   * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+   *   ContainerInterface.
+   */
+  public function __construct(LockBackendInterface $lock, LoggerChannelInterface $logger, ConfigFactory $configFactory, ContainerInterface $container) {
+    $this->lock = $lock;
+    $this->logger = $logger;
+    $this->configFactory = $configFactory;
+    $this->config = $this->configFactory->get('ymca_sync.settings');
```

Comment:
This is the only place we use $this->configFactory. We can simplify it to just 
$this->config = $configFactory->get('ymca_sync.settings');